### PR TITLE
[ENH] Improve TestAllGlobalForecasters

### PR DESCRIPTION
**Pull request by Xinyu-Wu-0000**
This PR reduces code duplication in `TestAllGlobalForecasters` and adds two new tests `test_global_forecasting_different_timestamps` and `test_global_forecasting_hier_series`.

### CODE REVIEW

**Comment by geetu040** [comment]
Does this PR incorporate the changes made here: https://github.com/sktime/sktime/pull/6666/files#diff-29b92508264b9a6103e8669be133f8f75d8a3b10a03461368e8d27b08fa329e0

**Comment by Xinyu-Wu-0000** [comment]
> Does this PR incorporate the changes made here: https://github.com/sktime/sktime/pull/6666/files#diff-29b92508264b9a6103e8669be133f8f75d8a3b10a03461368e8d27b08fa329e0

No, but a great idea to do it if this PR could be merged before #6666. I will add it.

**Comment by yarnabrina** [comment]
This is indeed the better place, thanks for suggesting @geetu040 .

@benHeid there are a lot of failure like this from `PyKANForecaster`, and does not look like sporadic failures. Can you please take a look?

> TypeError: MultKAN.__init__() got an unexpected keyword argument 'device'

**Comment by geetu040** [comment]
> @benHeid there are a lot of failure like this from `PyKANForecaster`, and does not look like sporadic failures. Can you please take a look?

They are coming from the latest pykan version which has been bounded in the latest code of main branch. So @Xinyu-Wu-0000 merging the main branch should fix these pykan related failures

**Comment by Xinyu-Wu-0000** [comment]
Merged.

**Review by yarnabrina — APPROVED** [review]
LGTM, but let's wait for @benHeid 's review who review global tests before.

File: sktime/forecasting/tests/test_all_forecasters.py [hunk 1]
```diff
@@ -931,101 +931,53 @@ def test_pridect_signature(self, estimator_class):
...
-    def test_global_forecasting_multiindex_hier(self, estimator_instance):
-        from sktime.utils._testing.hierarchical import _make_hierarchical
-
-        data_length = 100
-        data = _make_hierarchical(
-            (5, 100),
-            n_columns=2,
-            max_timepoints=data_length,
-            min_timepoints=data_length,
+    def _check_global_tag(self, estimator_instance):
```
Comment (benHeid) [inline sktime/forecasting/tests/test_all_forecasters.py:934]: I am not sure if this is a good idea. 

But could we create a fixture that is only returning global forecasters. Probably would require to override generate_estimator_instance? @fkiraly what is your opinion on this?

Furthermore, I am unsure if this should be part of this PR.

**Review by benHeid — COMMENTED** [review]
Looks good to me. I only have a small question if generating the correct forecaster based on fixtures would be better.

File: sktime/forecasting/tests/test_all_forecasters.py (hunk 1 above)
Comment (Xinyu-Wu-0000) [reply sktime/forecasting/tests/test_all_forecasters.py:934]: I tried to override `_generate_estimator_instance` but to find that `check_estimator` actually doesn't call it at all. 
Therefor if we want `check_estimator` to work, the tag checking has to be done inside the tests.

Comment (benHeid) [reply sktime/forecasting/tests/test_all_forecasters.py:934]: Oh I See. Only the test generation is calling it. Ok in this case. I agree in merging this branch.

**Review by benHeid — APPROVED** [review]

**Pull request merged**